src/isar/state_machine/state_machine.py

```diff
@@ -92,6 +180,84 @@ def __init__(
         self.transitions_log_length = transitions_log_length
         self.transitions_list: Deque[States] = deque([], self.transitions_log_length)
 
+    def _step_initiated(self) -> None:
+        self.current_step.status = StepStatus.InProgress
+        self.publish_step_status()
+        self.logger.info(
+            f"Successfully initiated "
+            f"{type(self.current_step).__name__} "
+            f"step: {str(self.current_step.id)[:8]}"
+        )
+
+    def _pause(self) -> None:
+        self.paused = True
+
+    def _unpause(self) -> None:
+        self.current_mission.status = MissionStatus.InProgress
+        self.current_task.status = TaskStatus.InProgress
+        if self.mqtt_client:
+            self.publish_mission_status()
+            self.publish_task_status()
```

Comment:
Having reviewed the code below it seems this if statement should no longer be required.